diagnostic_feedback/tests/integration/test_student_dg_quiz.py

```diff
@@ -0,0 +1,154 @@
+from .base_test import StudentBaseTest
+
+
+class StudentDiagnosticStyleTest(StudentBaseTest):
+
+    def test_for_single_grp_efficient_team(self):
+        self.load_student_view('dg_quiz', {"mode": "standard"})
+
+        next_btn = self._get_next_button()
+        back_btn = self._get_previous_button()
+        startover_link = self._get_startover_button_link()
+
+        self.assertEqual(startover_link.is_displayed(), False)
+
+        self._verify_question(0, next_btn, back_btn, 0)
+        self._submit_question()
+
+        self._verify_question(1, next_btn, back_btn, 0)
+        self._submit_question()
+
+        self._verify_question(2, next_btn, back_btn, 0)
+        self._submit_question(is_last=True)
+
+        self.wait_until_visible(startover_link)
+
+        self.assertEqual(startover_link.is_displayed(), True)
+
+        final_result = self.browser.find_element_by_css_selector('div.response_body').text
+
+        self.assertEqual('Efficient' in final_result, True)
+
+    def test_for_single_grp_mediocre_team(self):
+        self.load_student_view('dg_quiz', {"mode": "standard"})
+
+        next_btn = self._get_next_button()
+        back_btn = self._get_previous_button()
+        startover_link = self._get_startover_button_link()
+
+        self.assertEqual(startover_link.is_displayed(), False)
+
+        self._verify_question(0, next_btn, back_btn, 0)
+        self._submit_question()
+
+        self._verify_question(1, next_btn, back_btn, 1)
+        self._submit_question()
+
+        self._verify_question(2, next_btn, back_btn, 0)
+        self._submit_question(is_last=True)
+
+        self.wait_until_visible(startover_link)
+
+        self.assertEqual(startover_link.is_displayed(), True)
+
+        final_result = self.browser.find_element_by_css_selector('div.response_body').text
+
+        self.assertEqual('Mediocre' in final_result, True)
+
+    def test_for_single_grp_awful_team(self):
+        self.load_student_view('dg_quiz', {"mode": "standard"})
+
+        next_btn = self._get_next_button()
+        back_btn = self._get_previous_button()
+        startover_link = self._get_startover_button_link()
+
+        self.assertEqual(startover_link.is_displayed(), False)
+
+        self._verify_question(0, next_btn, back_btn, 2)
+        self._submit_question()
+
+        self._verify_question(1, next_btn, back_btn, 1)
+        self._submit_question()
+
+        self._verify_question(2, next_btn, back_btn, 2)
+        self._submit_question(is_last=True)
+
+        self.wait_until_visible(startover_link)
+
+        self.assertEqual(startover_link.is_displayed(), True)
+
+        final_result = self.browser.find_element_by_css_selector('div.response_body').text
```

Comment:
@jareerahsan Might be worth factoring the logic for finding the result text out into a separate method (something like `DiagnosticFeedbackBaseTest._get_final_result`), since all of the `test_for_single_group_*` methods in this file and in test_student_df_quiz.py are using it.